paper-server/src/main/java/org/bukkit/craftbukkit/Main.java

```diff
@@ -224,15 +230,20 @@ public static void main(String[] args) {
                 }
 
                 if (Main.class.getPackage().getImplementationVendor() != null && System.getProperty("IReallyKnowWhatIAmDoingISwear") == null) {
-                    Date buildDate = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss Z").parse(Main.class.getPackage().getImplementationVendor()); // Paper
+                    Date buildDate = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss Z").parse(Main.class.getPackage().getImplementationVendor());
 
                     Calendar deadline = Calendar.getInstance();
                     deadline.add(Calendar.DAY_OF_YEAR, -14);
+
+                    LocalDate localDate = LocalDate.now();
+                    LocalDate buildLocalDate = buildDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+                    Long daysSinceLastUpdate = ChronoUnit.DAYS.between(buildLocalDate, localDate);
+
                     if (buildDate.before(deadline.getTime())) {
-                        // Paper start - This is some stupid bullshit
-                        System.err.println("*** Warning, you've not updated in a while! ***");
-                        System.err.println("*** Please download a new build from https://papermc.io/downloads/paper ***");
-                        // Paper end
+                        System.err.println("*** It seems you have not updated Paper in a while ***");
```

Comment:
This message really isn't useful at all. Just remove it in my opinion. The message that says if the server is _actually_ out of date is the only useful one.

Reply:
should we make it check for updates on every startup in that case?